### ISSUE TITLE
SDL_MOUSEMOTION Y value always at maximum when a mouse rect is active

### ISSUE THREAD

**Issue by cgutman** [libsdl-org/SDL#4962]
When testing the new `SDL_SetWindowMouseRect()` functionality, I found some very odd behavior where the SDL_MOUSEMOTION event Y coordinate was always set to the boundary of the mouse rect regardless of the actual location of the cursor within the mouse rect.

A simple way to reproduce this is: `SDL_EVENT_LOGGING=2 ./test/testgles2 --confine-cursor 0,0,640,480`

the X value seems to change just fine, but Y remains stuck (even as `yrel` shows the mouse moving in the Y axis):
```
INFO: SDL EVENT: SDL_MOUSEMOTION (timestamp=6148 windowid=2 which=0 state=0 x=262 y=479 xrel=-32 yrel=-13)
INFO: SDL EVENT: SDL_MOUSEMOTION (timestamp=6166 windowid=2 which=0 state=0 x=243 y=479 xrel=-19 yrel=-16)
INFO: SDL EVENT: SDL_MOUSEMOTION (timestamp=6182 windowid=2 which=0 state=0 x=223 y=479 xrel=-20 yrel=-31)
INFO: SDL EVENT: SDL_MOUSEMOTION (timestamp=6198 windowid=2 which=0 state=0 x=208 y=479 xrel=-15 yrel=-36)
INFO: SDL EVENT: SDL_MOUSEMOTION (timestamp=6215 windowid=2 which=0 state=0 x=201 y=479 xrel=-7 yrel=-72)
```

I'm not sure if this is specific to the particular video backend, but I've noted this behavior with both X11 and Wayland backends on GNOME 41.1.

cc: @flibitijibibo

**Comment by flibitijibibo** [comment]
I think this commit added those events?

https://github.com/libsdl-org/SDL/commit/fd79607eb077d364b5e46b1e48436cf29c86eee9#diff-e844d0f1fbc6bd5610c387d3194f45e02860724e4f2a37b4c96b46c89ed7f59f

**Comment by cgutman** [comment]
Ah yep thanks, I couldn't understand how this same peculiar bug was impacting seemingly unrelated backends.

Fixed it in d7d67af44164fdf80c8f7103b22ad5bdb356dc7a
